packages/presence-signal/src/createPresenceSignal.ts

```diff
@@ -0,0 +1,108 @@
+import { createSignal, createEffect, onCleanup, type Accessor } from "solid-js";
+
+type SharedTransitionConfig = {
+  /** Duration in milliseconds used both for enter and exit transitions. */
+  transitionDuration: number;
+};
+
+type SeparateTransitionConfig = {
+  /** Duration in milliseconds used for enter transitions (overrides `transitionDuration` if provided). */
+  enterTransitionDuration: number;
+  /** Duration in milliseconds used for exit transitions (overrides `transitionDuration` if provided). */
+  exitTransitionDuration: number;
+};
+
+type Options = (
+  | SharedTransitionConfig
+  | SeparateTransitionConfig
+  | (SharedTransitionConfig & SeparateTransitionConfig)
+) & {
+  /** Opt-in to animating the entering of an element if `isVisible` is `true` during the initial mount. */
+  initialEnter?: boolean;
+};
+
+/**
+ * Animates the appearance of its children.
+ */
+export default function createPresenceSignal(
+  /** Indicates whether the component that the resulting values will be used upon should be visible to the user. */
+  isVisible: Accessor<boolean>,
+  optsValue: Options | Accessor<Options>,
+) {
+  const opts = () => (typeof optsValue === "function" ? optsValue() : optsValue);
+  const exitTransitionDuration = () => {
+    const optsValue = opts();
+    return "exitTransitionDuration" in optsValue
+      ? optsValue.exitTransitionDuration
+      : optsValue.transitionDuration;
+  };
+  const enterTransitionDuration = () => {
+    const optsValue = opts();
+    return "enterTransitionDuration" in optsValue
+      ? optsValue.enterTransitionDuration
+      : optsValue.transitionDuration;
+  };
+
+  const initialEnter = () => opts().initialEnter ?? false;
```

Comment:
Why is this a function? the state of `initialEnter` never change, right?

Reply:
Ah, this is just a side effect of my conversion over the original library. My methodology was to convert every piece of derived state into a function that returned that value. but that was entirely unecessary here as `initialEnter` is only ever used to initialize other signals. Fixed 👍